### PR TITLE
OI-1078 Add Edge support to IDE

### PR DESCRIPTION
**Pull request by savkaoleg**

### CODE REVIEW

**Review by romovs — CHANGES_REQUESTED** [review]

File: app/main/services/SeleniumService/edgedriver-versions.json [hunk 1]
```diff
@@ -0,0 +1,50 @@
+{
```
[the remark this answers is not in the document]
Comment (savkaoleg) [reply app/main/services/SeleniumService/edgedriver-versions.json:1]: done

File: app/main/services/SeleniumService/index.js [hunk 2]
```diff
@@ -295,6 +409,55 @@ export default class SeleniumService extends ServiceBase {
...
+                }
+            }
+        }
+
+        return result;
+    }
+
+    getEdgeDriverVersion(edgeVersion) {
+        return new Promise((resolve, reject) => {
+            // try getting the version from a local version map file
```
[the remark this answers is not in the document]
Comment (savkaoleg) [reply app/main/services/SeleniumService/index.js:429]: done

File: app/main/services/SeleniumService/index.js [hunk 3]
```diff
@@ -416,6 +610,22 @@ export default class SeleniumService extends ServiceBase {
         });
     }
 
+    // kill any active edgedriver processes
+    _killEdgeDriverProcess() {
+        return new Promise((resolve, reject) => {
+            if (process.platform === 'win32') {
+                const cp = spawn('taskkill', ['/IM', 'msedge.exe', '/F']);
```
[the remark this answers is not in the document]
Comment (savkaoleg) [reply app/main/services/SeleniumService/index.js]: done

File: app/main/services/SeleniumService/index.js [hunk 4]
```diff
@@ -416,6 +610,22 @@ export default class SeleniumService extends ServiceBase {
...
+                cp.on('exit', () => resolve(true));
+                // ignore errors since the process does not necessary exists at all
+                cp.on('error', () => resolve(true));
+            } else {
+                const cp = spawn('killall', ['msedge']);
```
[the remark this answers is not in the document]
Comment (savkaoleg) [reply app/main/services/SeleniumService/index.js]: done

File: app/renderer/components/dialogs/ChromeDriverDownloadingFailedDialog.js [hunk 5]
```diff
@@ -53,7 +53,7 @@ export default class ChromeDriverDownloadingFailedDialog extends React.PureCompo
                 <div>
                     <p>{'Download the appropriate driver for your version of Chrome from'} <a href='https://chromedriver.chromium.org/downloads' onClick={this.processLink}>here.</a></p>
                     <p>{'Extract the archive.'}</p>
-                    <p>{`Place the ChromeDriver executable directly into ${path}`}</p>
+                    <p>{`Place the ChromeDriver executable directly into ${path} with name: chromedriver-{versionname} folder`}</p>
```
[the remark this answers is not in the document]
Comment (savkaoleg) [reply app/renderer/components/dialogs/ChromeDriverDownloadingFailedDialog.js]: done

File: app/renderer/components/dialogs/EdgeDriverDownloadingFailedDialog.js [hunk 6]
```diff
@@ -0,0 +1,63 @@
...
+                        >
+                            Ok
+                        </Button>
+                    </Fragment>
+                )}
+            >
+                <div>
+                    <p>{'Download the appropriate driver for your version of Chrome from'} <a href='https://developer.microsoft.com/en-us/microsoft-edge/tools/webdriver/' onClick={this.processLink}>here.</a></p>
+                    <p>{'Extract the archive.'}</p>
+                    <p>{`Place the ChromeDriver executable directly into ${path} with name: edgedriver-{versionname} folder`}</p>
```
[the remark this answers is not in the document]
Comment (savkaoleg) [reply app/renderer/components/dialogs/EdgeDriverDownloadingFailedDialog.js]: done

File: package.json [hunk 7]
```diff
@@ -145,11 +145,11 @@
     "clean-stack": "2.2.0",
     "core-js": "3.6.4",
     "cross-spawn": "7.0.3",
-    "decompress-zip": "0.2.2",
     "detect-port": "1.3.0",
     "doctrine": "3.0.0",
     "electron-debug": "2.2.0",
     "electron-settings": "3.2.0",
+    "extract-zip": "^2.0.1",
```
[the remark this answers is not in the document]
Comment (savkaoleg) [reply package.json]: done

**Pull request merged**